### PR TITLE
CMR-9621: Humanizer cache to bootstrap

### PR DESCRIPTION
**Pull request by jmaeng72**
- Purposefully did not move humanizer cache to bootstrap because it is only refreshed within indexer APIs
- Moved humanizer-alias-cache to bootstrap so that we reduce the number of times this cache is refreshed. Before it was refreshed by all the nodes of ingest.

- Restructured humanizer-alias-cache to be hash-cache and updated all methods that call it. Humanizer alias cache used to be one string -> map of map of humanizer aliases. Now it is field-name -> map of aliases

Example:
	'platform' --> { 'TERRA' --> ['AM-1', 'am-1', 'AM 1'],  'OTHERPLATFORMS' --> ['otheraliases']}
	'tiling_system_name' --> {'TILE' --> ['tile_1', 'tile_2'], 'OTHERTILES' --> ['otheraliases']}
	'instrument' --> {'INSTRUMENT' --> ['instr1', 'instr2'], 'OTHERINSTRUMENTS' --> ['otheraliases']}"

### CODE REVIEW

File: ingest-app/src/cmr/ingest/validation/tiling_validation.clj [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
   "Provides functions to validate the tiling identification system names during collection update"
   (:require
-   [clojure.set :as s]
-   [clojure.string :as str]
-   [cmr.common.util :as util]
-   [cmr.ingest.services.humanizer-alias-cache :as humanizer-alias-cache]))
+	 [clojure.set :as s]
+	 [clojure.string :as str]
+	 [cmr.common.util :as util]
+	 [cmr.common-app.data.humanizer-alias-cache :as humanizer-alias-cache]))
```
Comment (jmaeng72) [inline ingest-app/src/cmr/ingest/validation/tiling_validation.clj:7]: On my IDE the formatting is different than what GitHub is showing here for all the validation files... I am going to try to fix it again and see what happens, but this is strange that the indent is showing up so off here...

File: bootstrap-app/src/cmr/bootstrap/api/routes.clj [hunk 2]
```diff
@@ -9,15 +9,16 @@
...
-   [cmr.common.api.context :as context]
+   [cmr.common-app.data.humanizer-alias-cache :as humanizer-alias-cache]
    [cmr.common-app.data.metadata-retrieval.collection-metadata-cache :as mc]
+   [cmr.common-app.data.search.collection-for-gran-acls-caches :as coll-for-gran-acls-caches]
    [cmr.common-app.services.kms-fetcher :as kms-fetcher]
    [cmr.common-app.services.provider-cache :as provider-cache]
    [cmr.common-app.services.search.elastic-search-index-names-cache :as elastic-search-index-names-cache]
-   [cmr.common-app.data.search.collection-for-gran-acls-caches :as coll-for-gran-acls-caches]
-   [cmr.common.api.errors :as errors]
    [cmr.common.log :refer [info]]
```
Comment (eereiter) [inline bootstrap-app/src/cmr/bootstrap/api/routes.clj:22]: Thanks for fixing the ordering.

File: common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  non-humanized-source-name-string --> list of aliases strings
+
+	Example:
+	Given: humanizer-field-name = 'platform'
+	Returns: { 'TERRA' --> ['AM-1', 'am-1', 'AM 1']
+						 'OTHERPLATFORMS' --> ['otheraliases']}"
+  [context humanizer-field-name]
+  (let [humanizer-alias-cache (hash-cache/context->cache context humanizer-alias-cache-key)
+        non-humanized-source-to-aliases-map (hash-cache/get-value humanizer-alias-cache humanizer-alias-cache-key humanizer-field-name)]
+				non-humanized-source-to-aliases-map))
```
Comment (eereiter) [inline common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj:99]: Line 97 through line 99 should be replaced by

(let [humanizer-alias-cache (hash-cache/context->cache context humanizer-alias-cache-key)]
   (hash-cache/get-value humanizer-alias-cache humanizer-alias-cache-key humanizer-field-name))

File: common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+(defconfig humanizer-alias-cache-job-refresh-rate
+           "Number of seconds between refreshes of the humanizer alias cache."
+           {:default 3600
+            :type Long})
+(defjob RefreshHumanizerAliasCache
```
Comment (eereiter) [inline common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj:105]: space between the defjob and defconfig

defconfig spacing is off.

File: common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        [ctx system]
+        (refresh-entire-cache {:system system}))
```
Comment (eereiter) [inline common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj:107]: defjob spacing is off.

File: common-app-lib/src/cmr/common_app/data/search/collection_for_gran_acls_caches.clj [hunk 6]
```diff
@@ -101,15 +101,15 @@
...
+  [context collection-concept-id]
    (let [collection-found (fetch-collections context collection-concept-id)
          coll-by-concept-id-cache (hash-cache/context->cache context coll-by-concept-id-cache-key)]
      (when-not (nil? collection-found)
        (hash-cache/set-value coll-by-concept-id-cache
                              coll-by-concept-id-cache-key
                              (:concept-id collection-found)
                              (clj-times->time-strs collection-found)))
-     collection-found)))
+     collection-found))
```
Comment (eereiter) [inline common-app-lib/src/cmr/common_app/data/search/collection_for_gran_acls_caches.clj:112]: Thanks for fixing.

File: ingest-app/src/cmr/ingest/services/humanizer_alias.clj [hunk 7]
```diff
@@ -187,14 +114,17 @@
                            plats)]
     (assoc collection plat-key updated-plats)))
 
+;; TODO test this path
```
Comment (eereiter) [inline ingest-app/src/cmr/ingest/services/humanizer_alias.clj:117]: Remove the TODO comment. I am assuming this path is tested.

**Review by eereiter — APPROVED** [review]
I approved it, but I have pointed out a couple of little items that need fixing.

File: ingest-app/src/cmr/ingest/services/humanizer_alias.clj (hunk 7 above)
Comment (jmaeng72) [reply ingest-app/src/cmr/ingest/services/humanizer_alias.clj]: done

File: common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj (hunk 5 above)
Comment (jmaeng72) [reply common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj]: done

File: common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj (hunk 4 above)
Comment (jmaeng72) [reply common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj:105]: done

File: common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj (hunk 3 above)
Comment (jmaeng72) [reply common-app-lib/src/cmr/common_app/data/humanizer_alias_cache.clj]: done

**Review by isja17 — APPROVED** [review]

**Comment by johnwteague** [comment]
You changed the cache definition so that it's now a redis-hash-cache right? Isn't that the main impact of the changes? (your description doesn't say)

**Pull request merged**